chitchat_server/models.py

```diff
@@ -36,18 +39,60 @@ class User(db.Model):
     admin: Mapped[bool]
     password: Mapped[str]
 
+    def __init__(self, email, name, admin, password_plaintext):
+        self.email = email
+        self.name = name
+        self.admin = admin
+        salt = secrets.token_hex(8)
+        salted_pass = salt[:8] + password_plaintext + salt[8:]
+        self.password = salt+hashlib.sha256(salted_pass.encode('ascii')).hexdigest()
+
+    def check_password(self, plaintext_password):
```

Comment:
This salting and hashing looks up to standard, it will protect the password in the database